File: Hackerrank_30daysOFcode_C#/day12.cs
using System;

class Person{
	protected string firstName;
	protected string lastName;
	protected int id;

	public Person(){}
	public Person(string firstName, string lastName, int identification){
			this.firstName = firstName;
			this.lastName = lastName;
			this.id = identification;
	}
	public void printPerson(){
		Console.WriteLine("Name: " + lastName + ", " + firstName + "\nID: " + id);
	}
}

class Student : Person{
    private int[] scores;

    public Student(String firstName, String lastName, int id, int[] scores) : base(firstName, lastName, id){
        this.firstName = firstName;
        this.lastName = lastName;
        this.id = id;
        this.scores = scores;
    }
    public Student(){}

    public String calculate(){
        int[] scores = this.scores;
        int s = 0;
        double a = 0;

        for(int i=0;i < scores.Length;i++){
            s += scores[i];
        }

        a = s / scores.Length;

        if (a >= 90 && a <= 100){
            return "O";
        }else if(a >= 80 && a < 90){
            return "E";
        }else if(a >= 70 && a < 80){
            return "A";
        }else if(a >= 55 && a < 70){
            return "P";
        }else if(a >= 40 && a < 55){
            return "D";
        }else{
            return "T";
        }
    }

}

class Solution {
	static void Main() {
		string[] inputs = Console.ReadLine().Split();
		string firstName = inputs[0];
	  	string lastName = inputs[1];
		int id = Convert.ToInt32(inputs[2]);
		int numScores = Convert.ToInt32(Console.ReadLine());
		inputs = Console.ReadLine().Split();
	  	int[] scores = new int[numScores];
		for(int i = 0; i < numScores; i++){
			scores[i]= Convert.ToInt32(inputs[i]);
		}

		Student s = new Student(firstName, lastName, id, scores);
		s.printPerson();
		Console.WriteLine("Grade: " + s.calculate() + "\n");
	}
}

File: 30daysOFcode_Java/day 12.java
import java.util.*;

class Person {
	protected String firstName;
	protected String lastName;
	protected int idNumber;
	
	// Constructor
	Person(String firstName, String lastName, int identification){
		this.firstName = firstName;
		this.lastName = lastName;
		this.idNumber = identification;
	}
	
	// Print person data
	public void printPerson(){
		 System.out.println(
				"Name: " + lastName + ", " + firstName 
			+ 	"\nID: " + idNumber); 
	}
	 
}
class Student extends Person{
    private int[] testScores;
    
    Student(String firstName, String lastName, int id, int[] testScores){
        super(firstName, lastName, id);
        this.testScores = testScores;
    }
    
    public char calculate(){
        int total = 0;
        double average;
        for(int i = 0; i < testScores.length; i++){
            total += testScores[i];
        }
        char ans;
        average = (double)total / testScores.length;
        if(average <= 100 && average >= 90.0){
            ans = "O".charAt(0);
        }else if(average < 90 && average >= 80){
            ans = "E".charAt(0);
        }else if(average < 80 && average >= 70){
            ans = "A".charAt(0);
        }else if(average < 70 && average >= 55){
            ans = "P".charAt(0);
        }else if(average < 55 && average >= 40){
            ans = "D".charAt(0);
        }else{
            ans = "T".charAt(0);
        }
        return ans;
    }
}



class Solution {
	public static void main(String[] args) {
		Scanner scan = new Scanner(System.in);
		String firstName = scan.next();
		String lastName = scan.next();
		int id = scan.nextInt();
		int numScores = scan.nextInt();
		int[] testScores = new int[numScores];
		for(int i = 0; i < numScores; i++){
			testScores[i] = scan.nextInt();
		}
		scan.close();
		
		Student s = new Student(firstName, lastName, id, testScores);
		s.printPerson();
		System.out.println("Grade: " + s.calculate() );
	}
}